include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -0,0 +1,382 @@
+
+					// fill cache with max values
+					auto fillCacheIfEnteredMaxValues = [&]()
+					{
+						const auto width = commonExecuteData.oit->imageExtent.width;
+						const auto height = commonExecuteData.oit->imageExtent.height;
+						core::vector4df_SIMD channels; // tmp, TODO
+
+						//convertColor<inFormat, outFormat>(srcPix, dstPix, blockX, blockY); nah		
+
+						if (localOutPos.X == width - 1 && localOutPos.Y == height - 1)
+						{
+							double finalMaxSumForTexel = {};
+
+							// encode and decode it properly
+							for (size_t yPos = 0; yPos < height; ++yPos)	// compute max sum values in rows for y texel sum values
+							{
+								// use finalMaxSumForTexel and pointer arythmetic to calculate final sum for a texel, towards LEFT
+								// gain max sum for X values per each y
+							}
+
+							for (size_t xPos = 0; xPos < width; ++xPos)
+							{
+								// use finalMaxSumForTexel and pointer arythmetic to calculate final sum for a texel, towards DOWN
+								// gain max sum for Y values per each X
+							}
+
+							if (!ExclusiveMode)
+							{
+								// add to finalMaxSumForTexel the reference texel as well
+							}
+						}
+					};
+
+					//uint8_t* imageSumOutData = nullptr; // TODO
+					//memcpy(commonExecuteData.outData+commonExecuteData.oit->getByteOffset(localOutPos,commonExecuteData.outByteStrides),commonExecuteData.inData+readBlockArrayOffset,commonExecuteData.outBlockByteSize);
+				};
+
+				CBasicImageFilterCommon::executePerRegion(commonExecuteData.inImg, sum, commonExecuteData.inRegions.begin(), commonExecuteData.inRegions.end(),clip);
+
+				return true;
+			};
+
+			// TODO return commonExecute(state, perOutputRegion);
+		}
+
+	private:
+
+		/*
+			Since it is desired to change the way local position are obtained, it can't use default common filter functions.
+		*/
+
+		template<typename F>
+		static inline void executePerRegion(const ICPUImage* image, F& f, const IImage::SBufferCopy* _begin, const IImage::SBufferCopy* _end)
+		{
+			CBasicImageFilterCommon::default_region_functor_t voidFunctor;
+			return executePerRegion<F, CBasicImageFilterCommon::default_region_functor_t>(image, f, _begin, _end, voidFunctor);
+		}
+
+		/*
+			The function will use default functor to determine which region it should take to execute operations on,
+			and that why the block will be specified - area of operations going on.
+		*/
+
+		template<typename F, typename G>
+		static inline void executePerRegion(const ICPUImage* image, F& f, const IImage::SBufferCopy* _begin, const IImage::SBufferCopy* _end, G& g)
+		{
+			for (auto it = _begin; it != _end; it++)
+			{
+				IImage::SBufferCopy region = *it;
+				if (g(region, it))
+					executePerBlock<F>(image, region, f);
+			}
+		}
+
+		/* 
+			Finally the function will be iterating through image passing it's local coordinates the way
+			a lambde being used can use it to fill preallocated table with data to compute the sum 
+		 */
+
+		template<typename F>
```

Comment:
these are in the base I'm telling you to inherit from

Reply:
corrected in following commit

Reply:
solved